From 0769546ba81df023fca4f86b3b0df46d88200607 Mon Sep 17 00:00:00 2001
From: bjorn3 <bjorn3@users.noreply.github.com>
Date: Tue, 26 May 2020 12:08:43 +0200
Subject: [PATCH] Remove windows 128bit int abi hack

---
 src/macros.rs | 18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/src/macros.rs b/src/macros.rs
index b02f3ea..d9a3475 100644
--- a/src/macros.rs
+++ b/src/macros.rs
@@ -165,24 +165,6 @@ macro_rules! intrinsics {
 
         $($rest:tt)*
     ) => (
-        #[cfg(all(windows, target_arch = "x86_64"))]
-        $(#[$($attr)*])*
-        pub extern $abi fn $name( $($argname: $ty),* ) -> $ret {
-            $($body)*
-        }
-
-        #[cfg(all(windows, target_arch = "x86_64"))]
-        pub mod $name {
-            #[cfg_attr(not(feature = "mangled-names"), no_mangle)]
-            pub extern $abi fn $name( $($argname: $ty),* )
-                -> ::macros::win64_128bit_abi_hack::U64x2
-            {
-                let e: $ret = super::$name($($argname),*);
-                ::macros::win64_128bit_abi_hack::U64x2::from(e)
-            }
-        }
-
-        #[cfg(not(all(windows, target_arch = "x86_64")))]
         intrinsics! {
             $(#[$($attr)*])*
             pub extern $abi fn $name( $($argname: $ty),* ) -> $ret {
-- 
2.20.1

